src/main/java/hudson/plugins/jira/JiraJobAction.java

```diff
@@ -83,19 +83,24 @@ public static void setAction(@Nonnull Job<?, ?> job, @Nonnull JiraSite site) thr
 
         // Find the first JIRA issue key in the branch name
         // If it exists, create the action and set it
+        Pattern pattern = site.getIssuePattern();
+        // Pipeline will URL encode job names if the branch or PR name contains a '/' or other non-URL safe characters
+        String decodedJobName = URLDecoder.decode(job.getName(), "UTF-8");
         String issueKey = null;
-        for (String part : StringUtils.split(job.getName(), '/')) {
-            Pattern pattern = site.getIssuePattern();
-            Matcher matcher = pattern.matcher(part);
-            if (matcher.matches() && matcher.groupCount() > 0) {
-                issueKey = matcher.group();
+        Matcher m = pattern.matcher(decodedJobName);
+        while (m.find()) {
+            if (m.groupCount() >= 1) {
+                issueKey = m.group(1);
+                break;
             }
         }
 
         if (issueKey != null) {
             JiraIssue issue = site.getIssue(issueKey);
-            job.addAction(new JiraJobAction(job, issue));
-            job.save();
+            if (issue != null) {
```

Comment:
There is a number of reasons this could happen, that's why it's good  to see it at least in INFO and it doesn't hurt the logs :)
- when no permissions to issue
- when issue was deleted by the admin
- when somebody made a typo